benchmark/src/main/java/com/linecorp/decaton/benchmark/ForkingExecution.java

```diff
@@ -50,67 +51,67 @@
 
     public static void main(String[] args) throws IOException, InterruptedException {
         Config config = mapper.readValue(args[0], Config.class);
+        RemoteCallback remoteCallback = BenchmarkRmi.lookup(Integer.parseInt(args[1]));
+
         InProcessExecution execution = new InProcessExecution();
-        BenchmarkResult result = execution.execute(config, System.out::println);
-        mapper.writeValue(System.out, result);
+        BenchmarkResult result = execution.execute(config, stage -> {
+            try {
+                remoteCallback.onStage(stage);
+            } catch (RemoteException e) {
+                log.error("Failed to invoke remote callback", e);
+                throw new RuntimeException(e);
+            }
+        });
+
+        remoteCallback.onResult(mapper.writeValueAsString(result));
```

Comment:
Any points to keep serializing result while we already have a different layer of object serialization? (RMI)

Reply:
Because we have to annotate `Serializable` for BenchmarkResult and all other inner classes.
Isn't better to serialize as JSON explicitly rather than forcing BenchmarkResult to have only Serializable fields (also for future extension)?

Reply:
ugh... that's Java it is.
I see. as we need to keep it serializable into JSON anyway (for json format output), I think it's fine to keep this as-is.